Code/Framework/AzCore/AzCore/Math/VectorN.h

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Internal/MathTypes.h>
+#include <AzCore/Math/Vector4.h>
+#include <AzCore/std/containers/vector.h>
+
+namespace AZ
+{
+    class ReflectContext;
+
+    //! N-dimensional vector class.
+    class VectorN final
+    {
+    public:
+
+        AZ_TYPE_INFO(VectorN, "{3C5A461A-3412-4D97-9CBC-856EE737B6DB}");
+
+        //! AzCore Reflection.
+        //! @param context reflection context
+        static void Reflect(ReflectContext* context);
+
+        VectorN() = default;
```

Comment:
A default constructor here is a only useful for serialization.
Because using it normally would result in the m_numElements being fixed to 0 since it is const.